pkgs/build-support/trivial-builders/default.nix

```diff
@@ -878,13 +878,17 @@ rec {
              ) + "-patched"
     , patches   ? []
     , postPatch ? ""
-    }: stdenvNoCC.mkDerivation {
+    , ...
+    }@args: stdenvNoCC.mkDerivation {
       inherit name src patches postPatch;
       preferLocalBuild = true;
       allowSubstitutes = false;
       phases = "unpackPhase patchPhase installPhase";
       installPhase = "cp -R ./ $out";
-    };
+    }
+    # Carry `meta` information from the underlying `src` if present.
+    // (optionalAttrs (builtins.hasAttr "meta" src) src.meta)
+    // (removeAttrs args [ "src" "name" "patches" "postPatch" ]);
```

Comment:
Quick note: I'm not sure how often this is actually used, but if this discussion re license mapping takes a little longer, we may want to consider pushing this part of the change to master already, I can imagine that there are more places where this would fix an issue with missing licenses.